### PR TITLE
[NUI] Deprecate visible instance fields [CA1051]

### PR DESCRIPTION
**Pull request by huiyueun**
CA1051: Do not declare visible instance fields
https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1051

Signed-off-by: huiyu.eun <huiyu.eun@samsung.com>

### API Changes ###

 - ACR: https://code.sec.samsung.net/jira/browse/TCSACR-396

### CODE REVIEW

**Comment by dongsug-song** [comment]
@huiyueun 
Hello,
please do not upload ACR-PR in DevelNUI branch.
here is a tip to handle ACR-PR here, I hope this helps.
thank you.

- Add-ACR : 
  1) the ACR-PR should be processed in master branch only.
  2) so, make API as hidden by adding `[EditorBrowsable(EditorBrowsableState.Never)]` attribute, then in DevelNUI branch API-Checker will pass.
  3) after the patch is merged into master branch by regular integration work, you can make new PR in master branch by removing `[EditorBrowsable(EditorBrowsableState.Never)]` attribute to proceed ACR.
  4) if the ACR is urgent, you can upload PR on master branch directly. that PR will be merged on DevelNUI branch by rebasing during the regular integration work.

- Deprecate-ACR : 
  1) the ACR-PR should be processed in master branch only.
  2) so, add some mark on the API by writing some comments like `//ToDo : will raise deprecated-ACR, [Obsolete("This has been deprecated in APIxxx and will be removed in APIxxx+2. The xxx variable is not used anymore and the xxx is replaced by xxx! Please use xxx instead!")`
  3) by step 2), the PR on DevelNUI will pass.
  4) after the patch is merged into master branch by regular integration work, you can make new PR in master branch by adding `[Obsolete("blah blah")]` tag in order to proceed ACR.
  5) if the ACR is urgent, you can upload PR on master branch directly. that PR will be merged on DevelNUI branch by rebasing during the regular integration work.

File: src/Tizen.NUI/src/public/VisualMaps.cs
Comment (m-macias) [inline src/Tizen.NUI/src/public/VisualMaps.cs:72]: It should be "Deprecated in API9, Will be removed in API11" instead. Also you should metion the alternatives if there are any to use.

Comment (huiyueun) [reply src/Tizen.NUI/src/public/VisualMaps.cs]: done

File: src/Tizen.NUI/src/internal/Disposable.cs
Comment (dongsug-song) [inline src/Tizen.NUI/src/internal/Disposable.cs:34]: Hello, 
I can see this is for Deprecated-ACR. 
please follow the steps below.
1) add comments like : `//ToDo : will raise deprecated-ACR, [Obsolete("Deprecated in API9, Will be removed in API11. Please use Disposed instead!")]`
2) get review in DevelNUI with No "ACR Required" tag
3) when it is merged in master branch by integration work, you can modify the comment of the [Obsolete] tag to be activated and use it for ACR process
4) the PR for ACR should be reviewed on master branch only

thank you.

File: src/Tizen.NUI/src/public/BaseHandle.cs
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/BaseHandle.cs:38]: Hello, 
I can see this is for Deprecated-ACR. 
please follow the steps below.
1) add comments like : `//ToDo : will raise deprecated-ACR, [Obsolete("Deprecated in API9, Will be removed in API11. Please use SwigCMemOwn instead!")]`
2) get review in DevelNUI with No "ACR Required" tag
3) when it is merged in master branch by integration work, you can modify the comment of the [Obsolete] tag to be activated and use it for ACR process
4) the PR for ACR should be reviewed on master branch only

thank you.

File: src/Tizen.NUI/src/public/BaseHandle.cs
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/BaseHandle.cs:47]: same as above.

File: src/Tizen.NUI/src/public/VisualMaps.cs
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/VisualMaps.cs:97]: same as above

**Review by dongsug-song — APPROVED** [review]
looks good to me
please merge this after the ACR is accepted.

File: src/Tizen.NUI/src/public/KeyValue.cs
Comment (hyunjushin) [inline src/Tizen.NUI/src/public/KeyValue.cs:36]: In this case, we can not provide replaceable property. It means application can not fix compile warning. If we can change the filed into Property in this patch, it is best. If we can't we can't deprecate the field.

File: src/Tizen.NUI/src/internal/Disposable.cs
Comment (huiyueun) [reply src/Tizen.NUI/src/internal/Disposable.cs]: I've modified to internal protected

File: src/Tizen.NUI/src/public/BaseHandle.cs
Comment (huiyueun) [reply src/Tizen.NUI/src/public/BaseHandle.cs]: I've modified to internal protected

File: src/Tizen.NUI/src/public/KeyValue.cs
Comment (huiyueun) [reply src/Tizen.NUI/src/public/KeyValue.cs:36]: I modified it by adding a setter/getter function. 
@dongsug-song @hyunjushin  How about these modifications?

File: src/Tizen.NUI/src/internal/Disposable.cs
Comment (dasg34) [reply src/Tizen.NUI/src/internal/Disposable.cs]: @huiyueun 
I think you need to add `[EditorBrowsable(EditorBrowsableState.Never)]` for `internal protected`. because derived calss can access `internal protected` types same as `protected` types.

File: src/Tizen.NUI/src/public/BaseHandle.cs
Comment (dasg34) [reply src/Tizen.NUI/src/public/BaseHandle.cs]: @huiyueun 
I think you need to add `[EditorBrowsable(EditorBrowsableState.Never)]` for `internal protected`. because derived calss can access `internal protected` types same as `protected` types.

File: src/Tizen.NUI/src/public/BaseHandle.cs
Comment (dasg34) [inline src/Tizen.NUI/src/public/BaseHandle.cs:576]: You can use expression body definition to implement a read-only property.
(https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/expression-bodied-members#read-only-properties)
```suggestion
        internal protected bool SwigCMemOwn => swigCMemOwn;
```

File: src/Tizen.NUI/src/public/BaseHandle.cs
Comment (dasg34) [inline src/Tizen.NUI/src/public/BaseHandle.cs:584]: You can use expression body definition to implement a read-only property.
(https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/expression-bodied-members#read-only-properties)
```suggestion
        internal protected bool Disposed => disposed;
```

File: src/Tizen.NUI/src/public/CustomViewRegistry.cs
Comment (dasg34) [inline src/Tizen.NUI/src/public/CustomViewRegistry.cs:114]: You can use expression body definition to implement a read-only property.
(https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/expression-bodied-members#read-only-properties)

```suggestion
        public ScriptableType Type => type;
```

File: src/Tizen.NUI/src/internal/Disposable.cs
Comment (huiyueun) [reply src/Tizen.NUI/src/internal/Disposable.cs]: @dasg34 ok, Thank you.

File: src/Tizen.NUI/src/public/BaseHandle.cs
Comment (huiyueun) [reply src/Tizen.NUI/src/public/BaseHandle.cs]: done

File: src/Tizen.NUI/src/public/BaseHandle.cs
Comment (huiyueun) [reply src/Tizen.NUI/src/public/BaseHandle.cs]: Done

File: src/Tizen.NUI/src/public/BaseHandle.cs
Comment (huiyueun) [reply src/Tizen.NUI/src/public/BaseHandle.cs]: done

File: src/Tizen.NUI/src/public/CustomViewRegistry.cs
Comment (huiyueun) [reply src/Tizen.NUI/src/public/CustomViewRegistry.cs]: done

File: src/Tizen.NUI/src/public/BaseHandle.cs
Comment (dasg34) [reply src/Tizen.NUI/src/public/BaseHandle.cs]: Thank you~

**Comment by dongsug-song** [comment]
ACR accepted

File: src/Tizen.NUI/src/internal/Disposable.cs
Comment (journey2w) [inline src/Tizen.NUI/src/internal/Disposable.cs:183]:
```suggestion
        /// The flag to check if it is already disposed of.
```

File: src/Tizen.NUI/src/public/BaseHandle.cs
Comment (journey2w) [inline src/Tizen.NUI/src/public/BaseHandle.cs:570]:
```suggestion
        /// The flag to check if it is already disposed of.
```

File: src/Tizen.NUI/src/public/CustomViewRegistry.cs
Comment (journey2w) [inline src/Tizen.NUI/src/public/CustomViewRegistry.cs:104]:
```suggestion
/// ScriptableType: Read Writable, non-animatable property, event thread only.
```

Line 90: 
```
/// Rhe enum of ScriptableType
```
=>
```
/// The enum of ScriptableType
```

File: src/Tizen.NUI/src/public/InputMethodContext.cs
Comment (journey2w) [inline src/Tizen.NUI/src/public/InputMethodContext.cs:947]:
```suggestion
                [Obsolete("Deprecated in API9, will be removed in API11, Use SwigCMemOwn which is declared in parent class")]
```

File: src/Tizen.NUI/src/public/KeyValue.cs
Comment (journey2w) [inline src/Tizen.NUI/src/public/KeyValue.cs:35]:
```suggestion
        [Obsolete("Deprecated in API9, will be removed in API11. Use IntegerKey instead.")]
```

**Review by journey2w — COMMENTED** [review]
I didn't add comments on the same descriptions repeatably.
Please all of them.

**Pull request merged**
